packages/terra-folder-tree/src/FolderTree.jsx

```diff
@@ -54,6 +55,72 @@ const FolderTree = ({
   intl,
 }) => {
   const folderTreeID = `folder-tree-${uuidv4()}`;
+  const listNode = useRef();
+
+  const handleKeyDown = event => {
+    const listItems = listNode.current.querySelectorAll('[data-item-show-focus=true]');
+    // Remove all hidden list items
+    const visibleListItems = Array.prototype.slice.call(listItems).filter((item) => {
+      let parent = item.parentNode;
+      while (parent && parent !== listNode.current) {
+        if (parent.hasAttribute('hidden')) {
+          return false;
+        }
+        parent = parent.parentNode;
+      }
+      return true;
+    });
+    const currentIndex = Array.from(visibleListItems).indexOf(event.target);
+    const lastIndex = visibleListItems.length - 1;
+
+    const handleHomeKey = () => visibleListItems[0].focus();
+    const handleEndKey = () => visibleListItems[lastIndex].focus();
+
+    switch (event.nativeEvent.keyCode) {
+      case KeyCode.KEY_END:
+        event.preventDefault();
```

Comment:
I'm probably missing the reason why, can we just call `event.preventDefault()` at the beginning of the `handleKeyDown` function to avoid calling it for every case?

Reply:
If you call it at the beginning, you won't be able to use any keys at all.

Reply:
I tried :( it expects a case statement to do so. And if I did something like
```
case true:
  event.preventDefault();
case KeyCode.KEY_RETURN:
case KeyCode.KEY_SPACE:
...
```

Our linter won't allow a fall-through to occur. It expects a break statement at the end of every case.

Reply:
Thank you for the explanation!